File: frontend/src/api/mitglied.ts
import { apiGet } from "./client";

export interface DienststundenSumme {
  funktion_id: number;
  funktion_name: string;
  summe_stunden: number;
  schwellenwert_stunden: number;
  schwellenwert_ueberschritten: boolean;
}

export interface MeinEinsatzKurz {
  id: number;
  titel: string;
  zeitpunkt: string;
}

export interface MitgliedUebersicht {
  einsaetze_jahr: number;
  dienste_jahr: number;
  dienststunden: DienststundenSumme[];
  letzte_einsaetze: MeinEinsatzKurz[];
}

export const holeMitgliedUebersicht = () =>
  apiGet<MitgliedUebersicht>("/mitglied/uebersicht");

File: backend/app/schemas/mitglied.py
from datetime import datetime

from pydantic import BaseModel

from app.schemas.dienststunden import DienststundenSummeOut


class MeinEinsatzKurz(BaseModel):
    id: int
    titel: str
    zeitpunkt: datetime


class MitgliedUebersicht(BaseModel):
    """Aggregierte eigene Kennzahlen für das persönliche Mitglieder-Dashboard."""

    einsaetze_jahr: int
    dienste_jahr: int
    dienststunden: list[DienststundenSummeOut]
    letzte_einsaetze: list[MeinEinsatzKurz]
